lib/https.js

```diff
@@ -31,9 +31,15 @@ const inherits = util.inherits;
 const debug = util.debuglog('https');
 const { urlToOptions, searchParamsSymbol } = require('internal/url');
 
-function Server(opts, requestListener) {
+function Server(opts = {}, requestListener) {
   if (!(this instanceof Server)) return new Server(opts, requestListener);
 
+  if (typeof opts === 'function') {
+    requestListener = opts;
+    opts = undefined;
```

Comment:
The problem is `opts = util._extend({}, opts);` shows it's not a consistent.

Do you mean that?

```js
...

const realOpts = util._extend({}, opts);
```